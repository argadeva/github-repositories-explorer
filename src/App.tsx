import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import FindUsers from './FindUsers'
import './App.css'

const queryClient = new QueryClient()

export default function App() {
  return (
    <QueryClientProvider client={queryClient}>
      <FindUsers />
    </QueryClientProvider>
  )
}
