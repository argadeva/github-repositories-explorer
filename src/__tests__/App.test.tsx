import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import App from '../App';
import { QueryClient } from '@tanstack/react-query';

// Mock FindUsers component
vi.mock('../FindUsers', () => ({
  default: () => <div data-testid="find-users">Mock FindUsers Component</div>
}));

describe('App Component', () => {
  it('should render without crashing', () => {
    const { container } = render(<App />);
    expect(container).toBeTruthy();
  });

  it('should render FindUsers component', () => {
    render(<App />);
    const findUsers = screen.getByTestId('find-users');
    expect(findUsers).toBeInTheDocument();
    expect(findUsers.textContent).toBe('Mock FindUsers Component');
  });

  it('should be wrapped in QueryClientProvider', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should initialize QueryClient with default config', () => {
    const queryClientSpy = vi.spyOn(QueryClient.prototype, 'mount');
    render(<App />);
    expect(queryClientSpy).toHaveBeenCalled();
    queryClientSpy.mockRestore();
  });

  it('should maintain component structure', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toBeInTheDocument();
    expect(screen.getByTestId('find-users')).toBeInTheDocument();
  });
});