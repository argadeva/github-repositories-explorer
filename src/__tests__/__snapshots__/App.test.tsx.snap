// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > should be wrapped in QueryClientProvider 1`] = `
<div
  data-testid="find-users"
>
  Mock FindUsers Component
</div>
`;
